grove/ising/ising_qaoa.py

```diff
@@ -0,0 +1,407 @@
+
+"""
+Finding the minimum energy for an Ising problem by QAOA.
+"""
+import pyquil.api as api
+from pyquil.paulis import exponential_map, PauliSum, PauliTerm
+from functools import reduce
+import pyquil.quil as pq
+from pyquil.gates import H
+from grove.pyvqe.vqe import VQE
+from scipy.optimize import minimize
+from collections import Counter
+from scipy import optimize
+import numpy as np
+import os
+import sys
+
+CXN = api.SyncConnection()
+
+
+class silence(object):
```

Comment:
Can I ask what's the intention for this? I'd like to know if there is a feature in the code somewhere that you would like to silence. If so we can work on removing that feature behind a feature flag, rather that trying to rerouting the output.

Reply:
I should probably have discussed this with you before implementing it. The aim of this code is to silence the printed strings ( e.g. "Parameters: [...]", "E => -1.0") during the optimization process of the VQE class from grove.pyvqe.vqe. I didn't want to edit a main code as VQE so I just coded this hack. I can either remove the "verbose" flag or we can add a flag to the VQE class.

Reply:
I think it would be better to add a `verbose` flag to VQE rather than catching stdout and re-routing it to /dev/null. Would you mind doing that?

Reply:
No problem, I will do that.